tests/performance-tests/src/auth/getAuthDetails.ts

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env node
+
+/* eslint-disable camelcase */
+import puppeteer, { Page } from 'puppeteer';
+
+export interface AuthTokens {
+  accessToken: string;
+  refreshToken: string;
+  expiryDate: Date;
+}
+
+export interface AuthDetails {
+  userName: string;
+  adminUrl: string;
+  supportsRefreshTokens: boolean;
+  authTokens: AuthTokens;
+}
+
+export type IdpOption = 'azure' | 'keycloak';
+
+export const getAuthTokens = async (
+  email: string,
+  password: string,
+  adminUrl: string,
+  idp: IdpOption,
+): Promise<{
+  id_token: string;
+  access_token: string;
+  refresh_token: string;
+  expires_at: Date;
+}> => {
+  const browser = await puppeteer.launch({
+    headless: true,
+    ignoreHTTPSErrors: true,
+    product: 'chrome',
+    executablePath: '/usr/bin/google-chrome',
```

Comment:
Would suggest creating a function to get the chrome executable path to cater for different environments i.e: 

```typescript
import os from 'os';


function getChromePath(): string {
  const platform = os.platform();

  switch(platform) {
    case 'win32':
      return 'C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe';

    case 'darwin':
      return '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome';

    default:
      return '/usr/bin/google-chrome';
  }
};

```

Reply:
This is actually the path that Chrome lives at inside the Docker container in which this script runs, but it's always possible that we wanna run this directly on our host machines, in which case the above would be useful, so I'm be happy to put it in!  Thanks for providing the snippet.

Reply:
Just switched "/usr/bin/chrome" for "/usr/bin/google-chrome" as that was the path in the container and on my local machine too.

Reply:
Ah right ok cool, keep forgetting this stuff runs purely through docker now